connect/eaas/runner/managers/base.py

```diff
@@ -51,6 +66,39 @@ def __init__(self, config: ConfigHelper, handler: EventsApp, enqueue):
         )
         self.running_tasks = 0
 
+    async def filter_collection_by_event_definition(self, client, task_data):
+        definition = self.config.event_definitions[task_data.input.event_type]
+        supported_statuses = []
+        template_args = {'_object_id_': task_data.input.object_id}
+        if 'events' in self.handler.get_features():
+            supported_statuses = self.handler.events[task_data.input.event_type]['statuses']
+            template_args['_statuses_'] = f'({",".join(supported_statuses)})'
+
+        rql_filter = Template(definition.api_collection_filter).substitute(
+            template_args,
+        )
+
+        collection = AsyncCollection(client, definition.api_collection_endpoint)
+
+        count = await collection.filter(rql_filter).count()
+        if count == 0:
+            logger.info(
+                f'Send skip response for {task_data.options.task_id} since '
+                'the current request status is not supported.',
+            )
+            self.send_skip_response(
+                task_data,
+                'The request status does not match the '
+                f'supported statuses: {",".join(supported_statuses)}.',
+            )
+
+        return count
+
+    def send_skip_response(self, data, output):
```

Comment:
probably this should be an abstract method in the base class so each subclass should use the right response class.

Reply:
Actually we can not make an abstract method, because not all subclasses use skip_response